pkg/steps/release/create_release.go

```diff
@@ -0,0 +1,139 @@
+package release
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	imageapi "github.com/openshift/api/image/v1"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	rbacclientset "k8s.io/client-go/kubernetes/typed/rbac/v1"
+
+	"github.com/openshift/ci-operator/pkg/api"
+	"github.com/openshift/ci-operator/pkg/steps"
+)
+
+// assembleReleaseStep will tag a full release suite
+// of images out to the configured namespace.
+type assembleReleaseStep struct {
+	config      api.ReleaseTagConfiguration
+	imageClient imageclientset.ImageV1Interface
+	podClient   steps.PodClient
+	rbacClient  rbacclientset.RbacV1Interface
+	jobSpec     *api.JobSpec
+}
+
+func (s *assembleReleaseStep) Inputs(ctx context.Context, dry bool) (api.InputDefinition, error) {
+	return nil, nil
+}
+
+func (s *assembleReleaseStep) Run(ctx context.Context, dry bool) error {
+	stable, err := s.imageClient.ImageStreams(s.jobSpec.Namespace).Get(api.StableImageStream, meta.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("could not resolve stable imagestream: %v", err)
+	}
+	cvo, ok := resolvePullSpec(stable, "cluster-version-operator")
+	if !ok {
+		log.Printf("No release image necessary, stable image stream does not include a cluster-version-operator image")
+		return nil
+	}
+	if _, ok := resolvePullSpec(stable, "cli"); !ok {
+		return fmt.Errorf("no 'cli' image was tagged into the stable stream, that image is required for building a release")
+	}
+
+	release, err := s.imageClient.ImageStreams(s.jobSpec.Namespace).Create(&imageapi.ImageStream{
+		ObjectMeta: meta.ObjectMeta{
+			Name: "release",
+		},
+	})
+	if err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return err
+		}
+		release, err = s.imageClient.ImageStreams(s.jobSpec.Namespace).Get("release", meta.GetOptions{})
```

Comment:
failed create --> eat error --> try a get

why? this "works" when the create error is a permissions error and the target is already exists?